### PR TITLE
[GCI] Add recipe for shanty

### PR DESCRIPTION
**Pull request by hometue**
Hi, please have a look at my work. Also, I have some doubts on the INSTALL() block, I am suppose to install it such that shanty can be run on the terminal by running the command "shanty", but not so sure if I did it correctly (or if I also did redundant stuff). Thanks.

### CODE REVIEW

File: app-misc/shanty/shanty-0.4.recipe [hunk 1]
```diff
@@ -0,0 +1,45 @@
+SUMMARY="Displays dialog boxes from command line and scripts under Haiku operating system"
```
Comment (humdingerb) [inline app-misc/shanty/shanty-0.4.recipe]: I suggest: "Dialog boxes from the command line and scripts"

File: app-misc/shanty/shanty-0.4.recipe [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+DESCRIPTION="Shanty is a program for displaying dialog boxes from command line and scripts\
```
Comment (humdingerb) [inline app-misc/shanty/shanty-0.4.recipe]: There's a space missing in front of the "\"

File: app-misc/shanty/shanty-0.4.recipe [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+under Haiku operating system. It is almost compatible with GNOME Zenity"
```
Comment (humdingerb) [inline app-misc/shanty/shanty-0.4.recipe]: Remove "operating system". Add a "." at the end.

File: app-misc/shanty/shanty-0.4.recipe [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+HOMEPAGE="https://github.com/peja/shanty"
+COPYRIGHT="2013, Kacper Kasper
+		2010, Milos Pejovic"
```
Comment (humdingerb) [inline app-misc/shanty/shanty-0.4.recipe]: Here and everywhere below, only one tab indention.

File: app-misc/shanty/shanty-0.4.recipe [hunk 5]
```diff
@@ -0,0 +1,45 @@
+SUMMARY="Displays dialog boxes from command line and scripts under Haiku operating system"
+DESCRIPTION="Shanty is a program for displaying dialog boxes from command line and scripts\
+under Haiku operating system. It is almost compatible with GNOME Zenity"
+HOMEPAGE="https://github.com/peja/shanty"
+COPYRIGHT="2013, Kacper Kasper
```
Comment (humdingerb) [inline app-misc/shanty/shanty-0.4.recipe]: No comma after the year.

File: app-misc/shanty/shanty-0.4.recipe [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+		2010, Milos Pejovic"
+LICENSE="MIT"
+REVISION="1"
+SOURCE_URI="https://github.com/peja/shanty/archive/f26aab0.zip"
+SOURCE_DIR="shanty-f26aab01709074147606a3699990ad713fc8702e"
```
Comment (humdingerb) [inline app-misc/shanty/shanty-0.4.recipe]: SOURCE_DIR under CHECKSUM

File: app-misc/shanty/shanty-0.4.recipe [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+CHECKSUM_SHA256="3db21622cea1e57aa7858572218b9e989d08e74005a84ea19e73f620c5fda5d3"
+
+ARCHITECTURES="?x86 x86_gcc2 ?x86_64"
+
+PROVIDES="
+		shanty = $portVersion
+		app:shanty = $portVersion
```
Comment (humdingerb) [inline app-misc/shanty/shanty-0.4.recipe]: Remove the app:shanty line.

File: app-misc/shanty/shanty-0.4.recipe [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+		cmd:shanty = $portVersion
+		"
+REQUIRES="
+		haiku
+		"
+
+BUILD_REQUIRES="
+		gcc
+		haiku_devel
```
Comment (humdingerb) [inline app-misc/shanty/shanty-0.4.recipe]: We usually have "haiku_devel" as first entry.

File: app-misc/shanty/shanty-0.4.recipe [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+		"
+
+BUILD()
+{
+	jam
+}
+
+INSTALL()
+{
+	mkdir -p $appsDir/shanty
```
Comment (humdingerb) [inline app-misc/shanty/shanty-0.4.recipe]: No need for $appsDir here. Just copy shanty to the $binDir

File: app-misc/shanty/shanty-0.4.recipe [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+		shanty = $portVersion
+		app:shanty = $portVersion
+		cmd:shanty = $portVersion
+		"
+REQUIRES="
+		haiku
+		"
+
+BUILD_REQUIRES="
+		gcc
```
Comment (humdingerb) [inline app-misc/shanty/shanty-0.4.recipe]: Move cmd:gcc to BUILD_PREREQUIRES:

File: app-misc/shanty/shanty-0.4.recipe [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+		haiku_devel
+		jam
```
Comment (humdingerb) [inline app-misc/shanty/shanty-0.4.recipe]: remove jam, it's already in BUILD_PREREQUIRES:

**Comment by hometue** [comment]
Hi, I made the changes, except for removing jam from "build_requires". Even if it is in "build_prerequires", the command "jam" just can't work (I ran with --no-dependencies). So I left it there.

**Comment by mmuman** [comment]
Oh, that's because there's a typo on BUILD_PREREQUIRES ;-) Here it does "gcc: command not found" btw.

**Comment by hometue** [comment]
Oh...whoops. Fixed it. I guess even while not writing a single line of code typos can still haunt me the same way.

**Comment by pulkomandy** [comment]
Some editors (at least vim and I think Pe) have syntax highlighting for recipe files. It may help to spot such problems.

**Comment by mmuman** [comment]
```
.git/rebase-apply/patch:52: new blank line at EOF.
+
warning: 1 line adds whitespace errors.
```
Please remove the last empty line, and I think we're good.

**Comment by hometue** [comment]
Fixed.
And pulkomandy, I kinda use notepad++ on my WIndows host and don't edit it in Haiku, running it in virtualbox without guest additions makes sharing files tough. I only edit it for the original few revisions to ensure it works and when I make changes that require me to test the recipe again (which admittedly brings various issues so I tend to test every edit now)

**Comment by mmuman** [comment]
Well, editing and doing git stuff from Haiku will save you from having to move files around. Of course you don't have a GUI git client, but that's an occasion to learn the command line. And it's not that slow in a VM. Someone should fix vboxsf someday but that's out of scope for GCI ;-)

**Comment by pulkomandy** [comment]
if your VM has network access, you can also use git (over ssh) to push/pull files to it.
Anyway, there is a Windows version of vim and you can import the syntax file there, or just convert it to Notepad++ syntax.
Vim syntax file for reference: http://cgit.haiku-os.org/haiku/tree/3rdparty/pulkomandy/recipe.syntax.vim

**Comment by hometue** [comment]
Well, I know the git commands, I use git bash even in Windows. But something seems to be wrong with network connectivity, it disconnects and reconnects periodically, making it so hard to download stuff at times. I am not very sure where does this issue lies so haven't gotten to fix it.

**Comment by mmuman** [comment]
Ah, you might want to try changing the emulated card to another type, the driver might work better.
Anyway, look good to me.

**Pull request merged**